### PR TITLE
Refactored GA actions to separate functions

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

Continues on the previous GA refactor PR

In this PR we
- create GA action functions
- separate GA actions in separate files per arias in the application
- define categories in one enum
- pull action messages to types objects so they are not spread all over the app but instead can be easily located in GA action files and modified there if necessary 

## To test
- Except for some smaller changes in wording of messages everything should work as previously

### CODE REVIEW

File: src/custom/utils/analytics/actions/lists.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    enable: 'Enable List',
+    disable: 'Disable List',
+  },
+  add: {
+    success: 'Add List Success',
+    failed: 'Add List Failed',
+  },
+}
+
+type updateListType = keyof typeof types.update
```
Comment (anxolin) [inline src/custom/utils/analytics/actions/lists.ts:23]: shouldn't types be UpperCamelCase?

**Review by anxolin — COMMENTED** [review]
I have to say, that although I like a lot that you divided by category, it feels all a bit over-engineered

Why dividing in so many files? 

utils.ts can live in `index.ts`. imo
same for `types.ts`

Then, i would also kill `actions/index.ts` you can re-export also in the `index.ts`
Also, the `actions` dir, is a good thing, but given that now there's only an `index` file around, we could have them all in the same analytics dir, making it easier to find. You can call them: `indexEvents.ts`, `listsEvents.ts`, etc.

Lastly the structure of the "actions", i see what you did there, and is kind of nice. 


But, for example, why 

```ts
const types = {
  remove: {
    start: 'Start Remove List',
    confirm: 'Confirm Remove List',
  },
  ...
}

type removeListType = keyof typeof types.remove
export function removeListAnalytics(option: removeListType, listUrl: string) {
  _reportEvent({
    category: Category.LIST,
    action: types.remove[option],
    label: listUrl,
  })
}
```
 
In my opinion, we can do:

```ts
type RemoveAction = 'Start' | 'Confirm'
export function removeListAnalytics(action: RemoveAction, listUrl: string) {
  _reportEvent({
    category: Category.LIST,
    action: `${action} Remove List`,
    label: listUrl,
  })
}
```

I found the second simpler to see close to the function what is the actual implementation.


Another example, just to show we don't always need to follow the same pattern. 

```ts
export function addListAnalytics(success: boolean, listURL: string) {
  _reportEvent({
    category: Category.LIST,
    action: 'Add List ' + (success? 'Success' : 'Failed'),
    label: listURL,
  })
}
```

**Comment by anxolin** [comment]
@nenadV91 did you review if our events match this doc? https://docs.google.com/spreadsheets/d/1AwcuvmPaAyIRuKayOEp1cKcVGD3MLOpfS8UWA04fBtY/edit#gid=0

File: src/custom/utils/analytics/listEvents.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { Category, _reportEvent } from './index'
+
+type UpdateListAction = 'Popup' | 'List Select'
```
Comment (anxolin) [inline src/custom/utils/analytics/listEvents.ts:3]: This is perfect, altough nitpicking `Popup` is not an action, its a location from which you take the action.
`UpdateListLocation`?

File: src/custom/utils/analytics/settingsEvents.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+export function showExpertModeConfirmationAnalytics() {
+  _reportEvent({
+    category: Category.EXPERT_MODE,
+    action: 'Show Confirmation',
+  })
+}
+
+export function toggleRecepientAddressAnalytics(isDisabling: boolean) {
```
Comment (anxolin) [inline src/custom/utils/analytics/settingsEvents.ts:18]: should we be coherent? if we add a flag, in line 4 you have `enable` and here is `isDisabling` 
We should probably go everywhere with `enable`?

File: src/custom/utils/analytics/themeEvents.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { Category, _reportEvent } from './index'
+
+export function toggleDarkModeAnalytics(darkMode: boolean) {
+  _reportEvent({
+    category: Category.THEME,
+    action: 'Toggle dark/light mode',
+    label: `${darkMode ? 'Light' : 'Dark'} mode`,
```
Comment (anxolin) [inline src/custom/utils/analytics/themeEvents.ts:7]: isn't this weird? : if darkMode --> light

Maybe name the param something that makes it more clear how to use it. 
`activateLightMode`?

File: src/custom/utils/analytics/transactionEvents.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
+import { Category, _reportEvent } from './index'
+
+type ExpirationAction = 'Default' | 'Custom'
```
Comment (anxolin) [inline src/custom/utils/analytics/transactionEvents.ts:3]: Again not `Action` but a `Type` or `Kind`

File: src/custom/utils/analytics/transactionEvents.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+export function orderExpirationTimeAnalytics(action: ExpirationAction, value: number) {
+  _reportEvent({
+    category: Category.ORDER_EXPIRATION_TIME,
+    action: `Set ${action} Expiration Time`,
+    value,
+  })
+}
+
+type wrapTransactionType = 'Send' | 'Sign' | 'Reject' | 'Error'
```
Comment (anxolin) [inline src/custom/utils/analytics/transactionEvents.ts:12]: why lowecase type?
Also, here is not a `Type` but is an `Action`

Comment (anxolin) [reply src/custom/utils/analytics/transactionEvents.ts:12]: U were not even consistent with `ClaimAction`

**Comment by anxolin** [comment]
Nice changes, I think it's all good other than some nitpicks. Looks great

File: src/custom/utils/analytics/transactionEvents.ts (hunk 6 above)
Comment (nenadV91) [reply src/custom/utils/analytics/transactionEvents.ts:12]: This one slipped through, I forgot to updated it. Good catch

**Review by anxolin — APPROVED** [review]

**Pull request merged**